packages/web-components/src/checkbox/checkbox.styles.ts

```diff
@@ -193,4 +193,39 @@ export const styles = css`
   :host([disabled]) .indeterminate-indicator {
     background-color: ${colorNeutralForegroundDisabled};
   }
-`;
+`.withBehaviors(
+  forcedColorsStylesheetBehavior(css`
+    .checked-indicator {
+      color: Highlight;
+    }
+    :host(:active) .indeterminate-indicator,
+    .indeterminate-indicator {
+      background-color: Highlight;
+    }
+    :host([aria-checked='true']) .control,
+    :host([aria-checked='mixed']) .control,
+    :host([aria-checked='true']:hover) .control,
+    :host([aria-checked='mixed']:hover) .control,
+    :host(:active) .control {
+      border-color: Highlight;
+    }
+    :host(:hover) .indeterminate-indicator {
+      background-color: Highlight;
+    }
+    :host([disabled]) .control,
+    :host([aria-checked='mixed'][disabled]) .control,
+    :host([aria-checked='true'][disabled]) .control {
```

Comment:
Same note here on :is()

```css
:host([disabled]) .control,
:host(:is([aria-checked='true'], [aria-checked='mixed'])[disabled]) .control {}
```

You may also not need that specificity...rarely should we use !important, but maybe that's doable with just the disabled attribute...